photutils/detection/tests/test_core.py

```diff
@@ -232,8 +239,16 @@ def test_no_peaks(self):
         tbl2 = find_peaks(data, 100000, centroid_func=centroid_com)
         assert set(tbl1.colnames) == set(tbl2.colnames)
 
-        tbl1 = find_peaks(data, 100, subpixel=True)
-        tbl2 = find_peaks(data, 100000, subpixel=True)
+        with catch_warnings(AstropyDeprecationWarning) as w:
+            tbl1 = find_peaks(data, 100, subpixel=True)
+
+        # find_peaks with subpixels uses deprecated cutout_footprint, thus
+        # 4 warnings are expected here
+        assert len(w) == 4
```

Comment:
Unfortunately, I am not familiar enough with the code base to know why out of the top of my head. I recommend insert a temporary commit to print out (via logger or print, whichever would show up in Travis log) and see where the extra two are coming from. It could very well be unrelated warnings from other packages that are triggered by the env used on Travis but not on your machine. 🤷‍♀️  Sorry that I can't be of more help.

Reply:
I also see this locally, but only when using `catch_warnings`. when running it in ipython/python only the expected number shows up. And even for other cases, `catch_warnings` catches the expected amount of warnings. 

Anyway, it's not critical for this, was more like a curiosity. If it comes up somewhere else I'll investigate.

Reply:
So, I had a look at this PR and `len(w) == 4` is indeed correct for the test. One warning for subpixel and three for footprints. This is because in `find_peaks`, that footprint function is called once per source and you have 3 sources in the `tbl1` result. All `catch_warnings` do modifying some settings for the same function name under `warnings`. So using Astropy's, you get all the warnings, but using Python default, it is only first occurrence of module + lineno (https://docs.python.org/3/library/warnings.html#the-warnings-filter). So, I am happy to report that all is well in this regard! 😄